app/src/test/js/keysCtrlTest.js

```diff
@@ -0,0 +1,139 @@
+describe('SyncKeysCtrl and SearchKeysCtrl', function() {
+
+    beforeEach(module('esbMessageAdminApp', function($provide) {
+        $provide.value("EsbMessageService", mockService);
+    }));
+
+    var syncKeysCtrl, searchKeysCtrl, rootScope;
+
+    beforeEach(inject(function($rootScope, $controller, _$q_) {
+        rootScope = $rootScope;
+        syncKeysScope = $rootScope.$new();
+        searchKeysScope = $rootScope.$new();
+        $q = _$q_;
+        syncKeysCtrl = $controller('SyncKeysCtrl', {
+            '$scope' : syncKeysScope
+        });
+        deferred.resolve(syncKeysSuccessResponse);
```

Comment:
If I understand how this is being used correctly, I would rather see specific service calls being mocked out instead of using a global `deferred` to resolve the next service call that happens. If that's not how this is working let me know. If that is what's happening then I don't think that's a good idea. It requires you to know implicitly what service call it's actually affecting, and order of calls may matter which is a little odd.

It would be easier to under the controller and how they use the service if you had something like below throughout the tests instead of using deferred.resolve:

```javascript
spyOn(mockService, "getSyncKeysTree").and.returnValue($q.when(syncKeysSuccessResponse));
```